frontend/src/pages/search.vue

```diff
@@ -1,160 +1,189 @@
+                no-virtual />
+            </v-window-item>
+            <v-window-item :key="5">
+              <item-grid
+                :items="personSearchResults"
+                :loading="loading"
+                no-virtual />
+            </v-window-item>
+            <v-window-item :key="6">
+              <item-grid
+                :items="artistSearchResults"
+                :loading="loading"
+                no-virtual />
+            </v-window-item>
+          </v-window>
         </v-col>
       </v-row>
     </v-container>
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { watch, computed, ref } from 'vue';
+import { useDebounceFn } from '@vueuse/core';
+import { useRoute, useRouter } from 'vue-router';
 import { BaseItemDto, BaseItemKind } from '@jellyfin/sdk/lib/generated-client';
 import { getPersonsApi } from '@jellyfin/sdk/lib/utils/api/persons-api';
 import { getItemsApi } from '@jellyfin/sdk/lib/utils/api/items-api';
-import { debounce } from 'lodash-es';
-import { useResponsiveClasses } from '@/composables';
+import { useRemote, useResponsiveClasses } from '@/composables';
+
+const route = useRoute();
+const router = useRouter();
+const remote = useRemote();
+
+const loading = ref(false);
+const searchTab = ref(0);
+const movieSearchResults = ref<BaseItemDto[]>([]);
+const showSearchResults = ref<BaseItemDto[]>([]);
+const albumSearchResults = ref<BaseItemDto[]>([]);
+const trackSearchResults = ref<BaseItemDto[]>([]);
+const bookSearchResults = ref<BaseItemDto[]>([]);
+const personSearchResults = ref<BaseItemDto[]>([]);
+const artistSearchResults = ref<BaseItemDto[]>([]);
+
+const searchQuery = computed(() => {
+  return route.query?.q?.toString() || '';
+});
 
-export default defineComponent({
-  setup() {
-    return { useResponsiveClasses };
-  },
-  data() {
-    return {
-      loading: false,
-      searchTab: 0,
-      topSearchResults: [] as BaseItemDto[],
-      movieSearchResults: [] as BaseItemDto[],
-      showSearchResults: [] as BaseItemDto[],
-      albumSearchResults: [] as BaseItemDto[],
-      trackSearchResults: [] as BaseItemDto[],
-      bookSearchResults: [] as BaseItemDto[],
-      personSearchResults: [] as BaseItemDto[],
-      artistSearchResults: [] as BaseItemDto[]
-    };
-  },
-  computed: {
-    searchQuery(): string {
-      return this.$route.query?.q?.toString() || '';
-    }
-  },
-  watch: {
-    searchQuery(newQuery: string): void {
-      if (newQuery !== '') {
-        this.performSearchDebounce();
-      }
-    }
-  },
-  async beforeMount() {
-    if (this.searchQuery === '') {
-      this.$router.back();
+const performSearchDebounce = useDebounceFn(
+  async () => await performSearch(),
+  500,
+  { maxWait: 2000 }
+);
+
+watch(
+  searchQuery,
+  async () => {
+    if (searchQuery.value !== '') {
+      await performSearchDebounce();
     } else {
-      await this.performSearch();
+      router.back();
```

Comment:
What do you think of putting the router.back in the performSearch? That way 1. logic is centralized 2. you can quickly erase your sentence and start typing again without being redirected back to the debounced logic.

Also you may want to use watchDebounced from vueuse?